### PR TITLE
fix(models): remove invalid sampling_metadata parameter from logits_processor

### LINKED ISSUE #4455: [Bug] TTS models pass removed sampling_metadata argument to LogitsProcessor
### Your current environment

<details>
<summary>The output of <code>python collect_env.py</code></summary>

```text
Your output of `python collect_env.py` here
```
==============================
        System Info
==============================
OS                           : Arch Linux (x86_64)
GCC version                  : (GCC) 16.1.1 20260430
Clang version                : 22.1.5
CMake version                : version 4.3.3
Libc version                 : glibc-2.43

==============================
       PyTorch Info
==============================
PyTorch version              : 2.11.0+cu130
Is debug build               : False
CUDA used to build PyTorch   : 13.0
ROCM used to build PyTorch   : N/A

==============================
      Python Environment
==============================
Python version               : 3.12.10 | packaged by conda-forge | (main, Apr 10 2025, 22:21:13) [GCC 13.3.0] (64-bit runtime)
Python platform              : Linux-7.0.10-arch1-1-x86_64-with-glibc2.43

==============================
       CUDA / GPU Info
==============================
Is CUDA available            : True
CUDA runtime version         : 13.2.78
CUDA_MODULE_LOADING set to   : 
GPU models and configuration : GPU 0: NVIDIA GeForce RTX 3060 Laptop GPU
Nvidia driver version        : 595.71.05
cuDNN version                : Could not collect
HIP runtime version          : N/A
MIOpen runtime version       : N/A
Is XNNPACK available         : True

==============================
          CPU Info
==============================
Architecture:                            x86_64
CPU op-mode(s):                          32-bit, 64-bit
Address sizes:                           48 bits physical, 48 bits virtual
Byte Order:                              Little Endian
CPU(s):                                  16
On-line CPU(s) list:                     0-15
Vendor ID:                               AuthenticAMD
Model name:                              AMD Ryzen 7 5800H with Radeon Graphics
CPU family:                              25
Model:                                   80
Thread(s) per core:                      2
Core(s) per socket:                      8
Socket(s):                               1
Stepping:                                0
Microcode version:                       0xa500012
Frequency boost:                         enabled
CPU(s) scaling MHz:                      58%
CPU max MHz:                             4465.2612
CPU min MHz:                             403.4880
BogoMIPS:                                6387.76
Flags:                                   fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_good nopl xtopology nonstop_tsc cpuid extd_apicid aperfmperf rapl pni pclmulqdq monitor ssse3 fma cx16 sse4_1 sse4_2 movbe popcnt aes xsave avx f16c rdrand lahf_lm cmp_legacy svm extapic cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw ibs skinit wdt tce topoext perfctr_core perfctr_nb bpext perfctr_llc mwaitx cpb cat_l3 cdp_l3 hw_pstate ssbd mba ibrs ibpb stibp vmmcall fsgsbase bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a rdseed adx smap clflushopt clwb sha_ni xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local user_shstk clzero irperf xsaveerptr rdpru wbnoinvd cppc arat npt lbrv svm_lock nrip_save tsc_scale vmcb_clean flushbyasid decodeassists pausefilter pfthreshold avic v_vmsave_vmload vgif v_spec_ctrl umip pku ospke vaes vpclmulqdq rdpid overflow_recov succor smca fsrm debug_swap
Virtualization:                          AMD-V
L1d cache:                               256 KiB (8 instances)
L1i cache:                               256 KiB (8 instances)
L2 cache:                                4 MiB (8 instances)
L3 cache:                                16 MiB (1 instance)
NUMA node(s):                            1
NUMA node0 CPU(s):                       0-15
Vulnerability Gather data sampling:      Not affected
Vulnerability Ghostwrite:                Not affected
Vulnerability Indirect target selection: Not affected
Vulnerability Itlb multihit:             Not affected
Vulnerability L1tf:                      Not affected
Vulnerability Mds:                       Not affected
Vulnerability Meltdown:                  Not affected
Vulnerability Mmio stale data:           Not affected
Vulnerability Old microcode:             Not affected
Vulnerability Reg file data sampling:    Not affected
Vulnerability Retbleed:                  Not affected
Vulnerability Spec rstack overflow:      Mitigation; Safe RET
Vulnerability Spec store bypass:         Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1:                Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:                Mitigation; Retpolines; IBPB conditional; IBRS_FW; STIBP always-on; RSB filling; PBRSB-eIBRS Not affected; BHI Not affected
Vulnerability Srbds:                     Not affected
Vulnerability Tsa:                       Mitigation; Clear CPU buffers
Vulnerability Tsx async abort:           Not affected
Vulnerability Vmscape:                   Mitigation; IBPB before exit to userspace

==============================
Versions of relevant libraries
==============================
[pip3] ema-pytorch==0.7.9
[pip3] flashinfer-python==0.6.12
[pip3] mypy==1.11.1
[pip3] mypy_extensions==1.1.0
[pip3] numpy==2.3.5
[pip3] nvidia-cublas==13.1.0.3
[pip3] nvidia-cuda-cccl==13.3.3.3.1
[pip3] nvidia-cuda-crt==13.3.33
[pip3] nvidia-cuda-cupti==13.0.85
[pip3] nvidia-cuda-nvcc==13.2.78
[pip3] nvidia-cuda-nvrtc==13.0.88
[pip3] nvidia-cuda-runtime==13.0.96
[pip3] nvidia-cuda-tileiras==13.2.78
[pip3] nvidia-cudnn-cu13==9.19.0.56
[pip3] nvidia-cudnn-frontend==1.25.0
[pip3] nvidia-cufft==12.0.0.61
[pip3] nvidia-cufile==1.15.1.6
[pip3] nvidia-curand==10.4.0.35
[pip3] nvidia-cusolver==12.0.4.66
[pip3] nvidia-cusparse==12.6.3.3
[pip3] nvidia-cusparselt-cu13==0.8.0
[pip3] nvidia-cutlass-dsl==4.5.2
[pip3] nvidia-cutlass-dsl-libs-base==4.5.2
[pip3] nvidia-cutlass-dsl-libs-cu13==4.5.2
[pip3] nvidia-ml-py==13.595.45
[pip3] nvidia-nccl-cu13==2.28.9
[pip3] nvidia-nvjitlink==13.0.88
[pip3] nvidia-nvshmem-cu13==3.4.5
[pip3] nvidia-nvtx==13.0.85
[pip3] nvidia-nvvm==13.2.78
[pip3] onnxruntime==1.24.4
[pip3] pyzmq==27.1.0
[pip3] tokenspeed-triton==3.7.10.post20260505
[pip3] torch==2.11.0+cu130
[pip3] torch_c_dlpack_ext==0.1.5
[pip3] torch-complex==0.4.4
[pip3] torchaudio==2.11.0+cu130
[pip3] torchcodec==0.13.0
[pip3] torchdiffeq==0.2.5
[pip3] torchmetrics==1.9.0
[pip3] torchsde==0.2.6
[pip3] torchvision==0.26.0+cu130
[pip3] transformers==5.8.1
[pip3] triton==3.6.0
[pip3] x-transformers==2.17.9
[conda] numpy                     2.3.2                    pypi_0    pypi

==============================
         vLLM Info
==============================
ROCM Version                 : Could not collect
vLLM Version                 : 0.23.0
vLLM-Omni Version            : 0.23.0rc2.dev7+g11829ccd7 (git sha: 11829ccd7)
vLLM Build Flags:
  CUDA Archs: Not Set; ROCm: Disabled
GPU Topology:
        GPU0    CPU Affinity    NUMA Affinity   GPU NUMA ID
GPU0     X      0-15    0               N/A

Legend:

  X    = Self
  SYS  = Connection traversing PCIe as well as the SMP interconnect between NUMA nodes (e.g., QPI/UPI)
  NODE = Connection traversing PCIe as well as the interconnect between PCIe Host Bridges within a NUMA node
  PHB  = Connection traversing PCIe as well as a PCIe Host Bridge (typically the CPU)
  PXB  = Connection traversing multiple PCIe bridges (without traversing the PCIe Host Bridge)
  PIX  = Connection traversing at most a single PCIe bridge
  NV#  = Connection traversing a bonded set of # NVLinks

==============================
     Environment Variables
==============================
CUDA_PATH=/opt/cuda
PYTORCH_NVML_BASED_CUDA_CHECK=1
TORCHINDUCTOR_COMPILE_THREADS=1
TORCHINDUCTOR_CACHE_DIR=/tmp/torchinductor_aja
LD_LIBRARY_PATH=/home/aja/vllm-omni/.venv/lib/python3.12/site-packages/cv2/../../lib64:

</details>

### Your code version

vLLM Version                 : 0.23.0
vLLM-Omni Version            : 0.23.0rc2.dev7+g11829ccd7 (git sha: 11829ccd7)

#### Bug Description

vLLM removed the `sampling_metadata` argument from `LogitsProcessor.__call__` in vllm-project/vllm#25345 ("[V0 Deprecation] Remove V0 sampling metadata").

The following vLLM-Omni TTS models still pass the removed argument:

* Higgs Audio V2
* Higgs Audio V3
* MOSS-TTS

As a result, `compute_logits()` raises a `TypeError` when invoking the current vLLM `LogitsProcessor`.

**Expected behavior**

These models should invoke the logits processor without `sampling_metadata`:

```python
self.logits_processor(lm_head, hidden_states)
```

instead of passing the removed argument.

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://vllm-omni.readthedocs.io), which can answer lots of frequently asked questions.

### PR DESCRIPTION
**Pull request by akshatvishu**
fixes  #4455

### CODE REVIEW

**Comment by hsliuustc0106** [comment]
**BLOCKING:**

**Test Coverage** — This PR fixes a bug but includes no regression test. Please add a test that:

1. Instantiates one of the affected models (e.g., MOSS-TTS)
2. Calls  with valid inputs
3. Verifies that the logits processor is invoked without errors

This will prevent similar breakage when vLLM makes API changes to LogitsProcessor in the future.

**Review by hsliuustc0106 — APPROVED** [review]

**Comment by linyueqian** [comment]
Reviewed as a requested reviewer. The fix is correct and complete, and I swept the other TTS/omni models for the same vLLM deprecation.

**Root cause.** vLLM removed `sampling_metadata` from `LogitsProcessor` in vllm-project/vllm#25345; the signature is now `forward(self, lm_head, hidden_states, embedding_bias=None)`. The old `self.logits_processor(self.lm_head, hidden_states, sampling_metadata)` binds a live `SamplingMetadata` to `embedding_bias`, which `_get_logits` passes as `bias=` into the lm_head `F.linear`, raising the `TypeError` in #4455. Dropping the third arg is the right fix.

**Completeness.** These three (`higgs_audio_v2`, `higgs_audio_v3`, `moss_tts`) were the only sites passing `sampling_metadata` positionally to `logits_processor`; everything else already uses the 2-arg form and `mimo_audio`'s call is commented out.

**Other models (same deprecation, different door).** The deprecation also dropped `sampling_metadata` from base-model `compute_logits` (now `compute_logits(self, hidden_states)`). A few models still forward `sampling_metadata` down a `compute_logits` chain, so I traced them: `ming_tts` (`MingLLMModel` `del`s it, then `Qwen2Model.compute_logits(hidden)` with 1 arg), `ming_flash_omni` (dropped in local `BailingMoeV2ForCausalLM` -> 2-arg `logits_processor`), `glm_tts` dit path (`GLMTTSDiTForGeneration.compute_logits` is a `return None`), and `dynin_omni` (base stage `del`s it). All safe: none reach a 3-positional `logits_processor` or a raw-vLLM `compute_logits`, and the dit/dynin stages run on the non-AR generation runner, which never calls `compute_logits`. No sibling fixes needed.

**Side note.** The AR runner wraps `compute_logits` in a `try/except TypeError` that retries without `sampling_metadata`, so this mismatch is currently absorbed there. Fixing the primary call (this PR) is preferable to leaning on that catch-all, which can also mask unrelated `TypeError`s.

**Test (@hsliuustc0106).** Agreed, and it can be weight-free, mirroring `tests/model_executor/models/moss_tts/test_per_request_codes.py`: build a bare talker via `__new__`, set `talker.logits_processor` to a stub that records its call, invoke `compute_logits(hidden, sampling_metadata=<sentinel>)`, and assert it was called with exactly `(lm_head, hidden)` and no third arg. Fails on `main`, passes here.

LGTM on the change.

**Comment by akshatvishu** [comment]
@hsliuustc0106  @linyueqian  Thanks for the review! As per the suggestion. I've added a new regression test to verify that `logits_processor` is called with only `(lm_head, hidden)`, without passing `sampling_metadata` for Moss-TTS!

I also couldn't find `tests/model_executor/models/moss_tts/test_per_request_codes.py` in this checkout, so I followed the same testing pattern directly in the new test.
